.circleci/config.yml

```diff
@@ -0,0 +1,173 @@
+          git clone -b master https://github.com/grackle-project/grackle $HOME/grackle
+          cd $HOME/grackle
+          ./configure
+          cd src/clib
+          make machine-linux-gnu
+          make
+          make install
+
+  install-docs-dependencies:
+    description: "Install dependencies for docs build."
+    steps:
+      - run: |
+          sudo apt-get update
+          sudo apt-get install -y dvipng texlive-latex-extra
+          python3 -m venv $HOME/venv
+          source $HOME/venv/bin/activate
+          pip install --upgrade pip
+          pip install --upgrade wheel
+          pip install --upgrade setuptools
+          pip install sphinx
+
+  download-test-data:
+    description: "Download test data."
+    steps:
+      - run: |
+          source $BASH_ENV
+          source $HOME/venv/bin/activate
+          girder-cli --api-url https://girder.hub.yt/api/v1 download 5af9ef42ec1bd30001fcd001 $HOME/enzo-dev/run/CosmologySimulation
+
+  build-and-test:
+    description: "Compile enzo and run tests."
+    parameters:
+      tag:
+        type: string
+        default: tip
+      skipfile:
+        type: string
+        default: ""
+      flags:
+        type: string
+        default: ""
+    steps:
+      - run: |
+          source $BASH_ENV
+          source $HOME/venv/bin/activate
+          if [ ! -f << parameters.skipfile >> ]; then
+            git checkout << parameters.tag >>
+            ./configure
+            cd src/enzo
+            make machine-linux-gnu
+            make load-config-test-suite
+            make clean
+            make -j 4
+            cd ../../run
+            python ./test_runner.py --suite=push -o $ENZOTEST_DIR --answer-name=$ANSWER_NAME --local --strict=high --verbose << parameters.flags >>
+          fi
+
+  build-docs:
+    description: "Test the docs build."
+    steps:
+      - run: |
+          source $HOME/venv/bin/activate
+          cd doc/manual/source
+          python -m sphinx -M html "." "_build" -W
+
+jobs:
+  test-suite:
+    docker:
+      - image: circleci/python:3.7.2
+
+    working_directory: ~/enzo-dev
+
+    steps:
+      - checkout
+      - set-env
+
+      - restore_cache:
+          name: "Restore dependencies cache."
+          key: dependencies-v1
+
+      - install-dependencies
+
+      - save_cache:
+          name: "Save dependencies cache"
+          key: dependencies-v1
+          paths:
+            - ~/.cache/pip
+            - ~/venv
+            - ~/local
+
+      - install-grackle
+      - download-test-data
+
+      - restore_cache:
+          name: "Restore test answers."
+          key: test-answers-v1
+
+      - build-and-test:
+          tag: tip
+          skipfile: ~/enzo_test/push_suite/push_suite.db
+          flags: --answer-store
+
+      - save_cache:
+          name: "Save test answers cache."
+          key: test-answers-v1
+          paths:
+            - ~/enzo_test/push_suite
+
+      - build-and-test:
+          tag: tip
```

Comment:
After this PR is accepted, this line should be changed to:
```
tag: gold-standard-v12
```
And then the appropriate tag added.

Reply:
@gregbryan, good question. CircleCI uses an immutable object store for caches, so if we want to change something or use updated dependencies, we have to increment the version number ourselves. Currently, only the `hypre` libraries are getting cached since we install everything else with `apt-get`.

Reply:
This system is good, but saying everything out loud makes me realize how many little pieces of information there are to remember. I'm going to make an issue about adding something to the developer documentation.